runtime/doc/lsp.txt

```diff
@@ -862,221 +1018,301 @@ workspace_symbol({query})                     *vim.lsp.buf.workspace_symbol()*
-start({cmd}, {cmd_args}, {handlers}, {extra_spawn_params})
-                Starts an LSP server process and create an LSP RPC client
-                object to interact with it.
+                Return: ~
+                    table list of diagnostics Helper function to ierate
+                    through diagnostic lines and return a position
+                    table {row, col} Move to the diagnostic position Get the
+                    previous diagnostic closes to the cursor_position
+                    M
+
+                            *vim.lsp.diagnostic.buf_get_prev_diagnostic_pos()*
+buf_get_prev_diagnostic_pos({bufnr}, {client_id}, {cursor_position})
+                Return the pos, {row, col}, for the prev diagnostic in the
+                current buffer.
 
                 Parameters: ~
-                    {cmd}                 (string) Command to start the LSP
-                                          server.
-                    {cmd_args}            (table) List of additional string
-                                          arguments to pass to {cmd}.
-                    {handlers}            (table, optional) Handlers for LSP
-                                          message types. Valid handler names
-                                          are:
-                                          • `"notification"`
-                                          • `"server_request"`
-                                          • `"on_error"`
-                                          • `"on_exit"`
-                    {extra_spawn_params}  (table, optional) Additional context
-                                          for the LSP server process. May
-                                          contain:
-                                          • {cwd} (string) Working directory
-                                            for the LSP server process
-                                          • {env} (table) Additional
-                                            environment variables for LSP
-                                            server process
+                    {bufnr}            number The buffer number
+                    {client_id}        number the client id
+                    {cursor_position}  table|nil See |nvim_win_get_cursor()|
+                                       or current cursor_position
 
                 Return: ~
-                    Client RPC object.
-                    Methods:
-                    • `notify()` |vim.lsp.rpc.notify()|
-                    • `request()` |vim.lsp.rpc.request()|
+                    table
 
-                    Members:
-                    • {pid} (number) The LSP server's PID.
-                    • {handle} A handle for low-level interaction with the LSP
-                      server process |vim.loop|.
+                               *vim.lsp.diagnostic.buf_move_next_diagnostic()*
+buf_move_next_diagnostic({bufnr}, {client_id}, {cursor_position})
```

Comment:
I personally feel that `goto` is a better word than `move`. But I am not native English.